pkg/proxy/iptables/proxier.go

```diff
@@ -278,11 +278,6 @@ func NewProxier(ipt utiliptables.Interface,
 	masqueradeValue := 1 << uint(masqueradeBit)
 	masqueradeMark := fmt.Sprintf("%#08x/%#08x", masqueradeValue, masqueradeValue)
 
-	if nodeIP == nil {
```

Comment:
This has been present for ~4 years (https://github.com/kubernetes/kubernetes/commit/1acaa1db09e96d297255270ecb1cf29791615fdf#diff-d51765b83fe795b469e8a86276b12dc9R218-R221). Can you explain why this is correct to remove?

Reply:
@liggitt I think https://github.com/kubernetes/kubernetes/pull/77167 introduced a breaking change where kube-proxy exits if nodeIP is nil. Before v1.16, if nodeIP is nil kube-proxy continues assuming later it would default node IP to 127.0.0.1.

Reply:
> This has been present for ~4 years ([1acaa1d#diff-d51765b83fe795b469e8a86276b12dc9R218-R221](https://github.com/kubernetes/kubernetes/commit/1acaa1db09e96d297255270ecb1cf29791615fdf#diff-d51765b83fe795b469e8a86276b12dc9R218-R221)). Can you explain why this is correct to remove?

At first, I thought that `iptables.NewProxier` was only used here, if so, it might be redundant to validate `nodeIP` again. Later, I found that it was used by https://github.com/kubernetes/kubernetes/blob/master/pkg/kubemark/hollow_proxy.go#L82, so I withdrew the modification here

Reply:
How is this different from simply rolling back #77167 ?  I thought the point of that was to force kube-proxy not to proceed with a wrong IP address?

Reply:
https://github.com/kubernetes/kubernetes/pull/77167 also adds exponential back-off which fixes https://github.com/kubernetes/kubernetes/issues/37414 where kube-proxy would try to get the node IP before the node is registered. IMO removing the nodeIP check entirely and having the proxy implementations fall back to use 127.0.0.1 is better.